src/role-permission-operator.ts

```diff
@@ -45,55 +48,110 @@ const NAMESPACE = Config.WATCH_NAMESPACE
 
 const roleResourceStore = new RoleResources()
 const oryKeto = new KetoTuples()
-const rolePermissionChangeProcessor = new KetoChangeProcessor(oryKeto.updateAllRolePermissions.bind(oryKeto))
+const ketoChangeProcessor = KetoChangeProcessor.getInstance()
+const updateKetoFn = async (fnArgs: any) => {
+  const boundedFn = oryKeto.updateAllRolePermissions.bind(oryKeto)
+  boundedFn(fnArgs.subjectObjectCombos)
+  logger.info('Updated the relation tuples in Keto', fnArgs.subjectObjectCombos)
+}
 
 const kc = new k8s.KubeConfig()
 kc.loadFromDefault()
 
+let healthStatus = "Unknown"
+
 // If we want to do something in K8S, we can use the following APIs
 // const k8sApi = kc.makeApiClient(k8s.AppsV1Api);
 // const k8sApiMC = kc.makeApiClient(k8s.CustomObjectsApi);
 // const k8sApiPods = kc.makeApiClient(k8s.CoreV1Api);
+const k8sApiCustomObjects = kc.makeApiClient(k8s.CustomObjectsApi)
 
 // Listen for events or notifications and act accordingly
 const watch = new k8s.Watch(kc)
 
 async function onEvent(phase: string, apiObj: any) {
   logger.info(`Received event in phase ${phase} for the resource ${apiObj?.metadata?.name}`)
-  // try {
-  //   const result = await k8sApiMC.listNamespacedCustomObject(
-  //     RESOURCE_GROUP,
-  //     RESOURCE_VERSION,
-  //     NAMESPACE,
-  //     RESOURCE_PLURAL,
-  //   )
-  //   console.log(JSON.stringify(result.body, null, 2))
-  //   console.log(apiObj?.spec)
-  // } catch (err: any) {
-  //   logger.error(err);
-  // }
   const resourceName = apiObj?.metadata?.name
   const role = apiObj?.spec?.role
   const permissions = apiObj?.spec?.permissions
+  const generation = apiObj?.metadata?.generation + ''
+  // Ignore status updates by comparing generation number
+  if (phase === 'MODIFIED' && roleResourceStore.checkHash(resourceName, generation)) {
+    return
+  }
   if (resourceName && role && permissions) {
-    if (phase == "ADDED") {
-      roleResourceStore.updateRoleResource(resourceName, role, permissions)
-    } else if (phase == "MODIFIED") {
-      roleResourceStore.updateRoleResource(resourceName, role, permissions)
-    } else if (phase == "DELETED") {
+    if (phase === 'ADDED' || phase === 'MODIFIED') {
+      // Get the temporary consolidated role assignments based on already stored roles
+      const consolidatedRolePermissions = roleResourceStore.getConsolidatedTempData(resourceName, role, permissions)
+      const rolePermissions: RolePermissions[] = Object.entries(consolidatedRolePermissions).map(item => {
+        return {
+          rolename: (<any>item[1]).role,
+          permissions: (<any>item[1]).permissions
+        }
+      })
```

Comment:
Probably a bit better as
```suggestion
      const rolePermissions: RolePermissions[] = Object.values(consolidatedRolePermissions).map((item: any) => ({
        rolename: item.role,
        permissions: item.permissions
      }))
```

Reply:
Moreover, if possible, it would be better to give `getConsolidatedTempData` a meaningful return type.